pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -430,6 +430,11 @@ func (c *Cloud) CreateMasterPool(p model.MasterPool) error {
 		return err
 	}
 
+	validMachine := validateMachineType(p.MachineType)
```

Comment:
CLI does not care or know about valid AWS-specific machine types. There is also an idea to have a RESTful API instead of CLI.